client/asset/dcr/coin_selection.go

```diff
@@ -0,0 +1,210 @@
+		total := sum + toAtoms(unspent.rpc.Amount)
+		txFee := uint64(baseTxSize+inputSize+unspent.input.Size()) * feeRate
+		req := amt
+		if !subtract { // add the fee to required
+			req += txFee
+		}
+		if total < req {
+			return false, 0
+		}
+		excess := total - req
+		if subtract && excess > txFee {
+			excess -= txFee // total - (req + txFee), without underflow
+		}
+		if !reportChange || dexdcr.IsDustVal(dexdcr.P2PKHOutputSize, excess, feeRate) {
+			excess = 0
+		}
+		return true, excess
+	}
+}
+
+// orderEnough generates a function that can be used as the enough argument to
+// the fund method. If change from a split transaction will be created AND
+// immediately available (not mixing), set reportChange to indicate this and the
+// returned enough func will return a non-zero excess value reflecting this
+// potential spit tx change. Otherwise, the enough func will always return 0,
+// leaving only unselected UTXOs to cover any required reserves.
+func orderEnough(val, lots, feeRate uint64, reportChange bool) func(sum uint64, size uint32, unspent *compositeUTXO) (bool, uint64) {
+	return func(sum uint64, size uint32, unspent *compositeUTXO) (bool, uint64) {
+		reqFunds := calc.RequiredOrderFundsAlt(val, uint64(size+unspent.input.Size()), lots,
+			dexdcr.InitTxSizeBase, dexdcr.InitTxSize, feeRate)
+		total := sum + toAtoms(unspent.rpc.Amount) // all selected utxos
+
+		if total >= reqFunds { // that'll do it
+			// change = total - (val + swapTxnsFee)
+			excess := total - reqFunds // reqFunds = val + swapTxnsFee
+			if !reportChange || dexdcr.IsDustVal(dexdcr.P2PKHOutputSize, excess, feeRate) {
+				excess = 0
+			}
+			return true, excess
+		}
+		return false, 0
+	}
+}
+
+func sumUTXOs(set []*compositeUTXO) (tot uint64) {
+	for _, utxo := range set {
+		tot += toAtoms(utxo.rpc.Amount)
+	}
+	return tot
+}
+
+// In the following utxo selection functions, the compositeUTXO slice MUST be
+// sorted in ascending order (smallest first, largest last).
+
+// subsetLargeBias begins by summing from the largest UTXO down until the sum is
+// just below the requested amount. Then it picks the (one) final UTXO that
+// reaches the amount with the least excess. Each utxo in the input must be
+// smaller than amt - use via leastOverFund only!
+func subsetLargeBias(amt uint64, utxos []*compositeUTXO) []*compositeUTXO {
+	// Add from largest down until sum is *just under*. Resulting sum will be
+	// less, and index will be the next (smaller) element that would hit amt.
+	var sum uint64
+	var i int
+	for i = len(utxos) - 1; i >= 0; i-- {
+		this := toAtoms(utxos[i].rpc.Amount) // must not be >= amt
+		if sum+this >= amt {
+			break
+		}
+		sum += this
+	}
+
+	if i == -1 { // full set used
+		// if sum >= amt { return utxos } // would have been i>=0 after break above
+		return nil
+	}
+	// if i == len(utxos)-1 { return utxos[i:] } // shouldn't happen if each in set are < amt
```

Comment:
It's commented out here of course, but I'm not sure why I had written this as `utxos[i:]`, because that's the same as `utxos[i]` when `i == len(utxos)-1` (the last one).  As @martonp pointed out, it's not going to happen that just the one UTXO is sufficient because: `Each utxo in the input must be smaller than amt - use via leastOverFund only!` so this is commented.  Might as well remove.